### ISSUE TITLE
PayPal Vault - 422 (Unprocessable Entity)

### ISSUE THREAD

**Issue by terrifickid** [braintree/braintree-web#332]
### General information

* SDK version: JS V3 SDK
* Environment: Sandbox
* Browser and OS: Chrome Version 61.0.3163.100 (Official Build) (64-bit) / MacOS 10.12.6 

### Issue description
Using paypalCheckoutInstance.createPayment seems fine with flow:'checkout' - however my project requires flow:'vault'.

In the case of flow:'vault' I get `POST https://api.sandbox.braintreegateway.com/merchants/kcsnqyw983ggxndr/client_api/v1/paypal_hermes/setup_billing_agreement 422 (Unprocessable Entity)
i @ client.min.js:1
a @ client.min.js:1
t.exports @ client.min.js:1
(anonymous) @ client.min.js:1
r.request @ client.min.js:1
r.createPayment @ paypal-checkout.min.js:1
(anonymous) @ paypal-checkout.min.js:1
payment @ (index):71
(anonymous) @ hacks.js:88
ZalgoPromise @ promise.js:42
(anonymous) @ hacks.js:57
ZalgoPromise.try @ promise.js:380
(anonymous) @ component.jsx:283
resultValue @ props.js:85
(anonymous) @ serialize.js:37
ZalgoPromise.try @ promise.js:380
(anonymous) @ serialize.js:36
(anonymous) @ types.js:65
ZalgoPromise.try @ promise.js:380
_RECEIVE_MESSAGE_TYPE.(anonymous function) @ types.js:53
receiveMessage @ index.js:112
messageListener @ index.js:138
postMessage (async)
(anonymous) @ strategies.js:56
SEND_MESSAGE_STRATEGIES.(anonymous function) @ strategies.js:55
(anonymous) @ index.js:72
ZalgoPromise.try @ promise.js:380
(anonymous) @ index.js:66
(anonymous) @ promise.js:357
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:357
ZalgoPromise.map @ promise.js:364
(anonymous) @ index.js:64
ZalgoPromise.try @ promise.js:380
sendMessage @ index.js:29
(anonymous) @ client.js:133
ZalgoPromise @ promise.js:42
(anonymous) @ client.js:111
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ client.js:109
ZalgoPromise.try @ promise.js:380
request @ client.js:34
_send @ client.js:211
wrapper @ serialize.js:155
(anonymous) @ promise.js:40
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:39
resultValue @ props.js:85
(anonymous) @ fn.js:69
queryParam @ component.js:200
(anonymous) @ props.js:181
ZalgoPromise.try @ promise.js:380
getQueryParam @ props.js:179
(anonymous) @ props.js:233
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ props.js:226
propsToQuery @ props.js:204
ParentComponent.buildUrl @ index.js:450
buildUrl:memoized @ decorators.js:13
(anonymous) @ index.js:190
ZalgoPromise.try @ promise.js:380
ParentComponent.tryInit @ index.js:1250
ParentComponent.render @ index.js:120
(anonymous) @ index.js:253
ZalgoPromise.try @ promise.js:380
ParentComponent.tryInit @ index.js:1250
ParentComponent.renderTo @ index.js:231
(anonymous) @ index.js:473
ZalgoPromise.try @ promise.js:380
Component.renderTo @ index.js:472
(anonymous) @ popupBridge.js:172
doRender @ popupBridge.js:162
Checkout.renderTo @ popupBridge.js:172
callOriginal @ util.js:279
(anonymous) @ hacks.js:131
obj.(anonymous function) @ util.js:275
renderCheckout @ checkout.js:80
clickButton @ button.js:48
(anonymous) @ button.js:63
(index):104 checkout.js error Error: Error: PayPal payment options are invalid.
    at Object._RECEIVE_MESSAGE_TYPE.(anonymous function) [as postrobot_message_response] (https://www.paypalobjects.com/api/checkout.js:2863:118)
    at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2765:77)
    at messageListener (https://www.paypalobjects.com/api/checkout.js:2786:13)
    at Object._RECEIVE_MESSAGE_TYPE.(anonymous function) [as postrobot_message_response] (https://www.paypalobjects.com/api/checkout.js:2863:118)
    at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2765:77)
    at messageListener (https://www.paypalobjects.com/api/checkout.js:2786:13)
    at deserializeError (https://www.paypalobjects.com/api/checkout.js:3366:20)
    at https://www.paypalobjects.com/api/checkout.js:3385:322
    at https://www.paypalobjects.com/api/checkout.js:3471:30
    at eachArray (https://www.paypalobjects.com/api/checkout.js:3454:51)
    at each (https://www.paypalobjects.com/api/checkout.js:3460:35)
    at replaceObject (https://www.paypalobjects.com/api/checkout.js:3470:13)
    at https://www.paypalobjects.com/api/checkout.js:3472:169
    at eachObject (https://www.paypalobjects.com/api/checkout.js:3457:65)
    at each (https://www.paypalobjects.com/api/checkout.js:3460:144)
    at replaceObject (https://www.paypalobjects.com/api/checkout.js:3470:13)
onError @ (index):104
(anonymous) @ promise.js:40
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:39
resultValue @ props.js:85
(anonymous) @ fn.js:34
(anonymous) @ serialize.js:37
ZalgoPromise.try @ promise.js:380
(anonymous) @ serialize.js:36
(anonymous) @ types.js:65
ZalgoPromise.try @ promise.js:380
_RECEIVE_MESSAGE_TYPE.(anonymous function) @ types.js:53
receiveMessage @ index.js:112
messageListener @ index.js:138
postMessage (async)
(anonymous) @ strategies.js:56
SEND_MESSAGE_STRATEGIES.(anonymous function) @ strategies.js:55
(anonymous) @ index.js:72
ZalgoPromise.try @ promise.js:380
(anonymous) @ index.js:66
(anonymous) @ promise.js:357
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:357
ZalgoPromise.map @ promise.js:364
(anonymous) @ index.js:64
ZalgoPromise.try @ promise.js:380
sendMessage @ index.js:29
(anonymous) @ client.js:133
ZalgoPromise @ promise.js:42
(anonymous) @ client.js:111
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ client.js:109
ZalgoPromise.try @ promise.js:380
request @ client.js:34
_send @ client.js:211
wrapper @ serialize.js:155
props.(anonymous function) @ hacks.js:126
(anonymous) @ promise.js:40
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:39
resultValue @ props.js:85
(anonymous) @ fn.js:34
(anonymous) @ index.js:1287
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
ParentComponent.error @ index.js:1284
(anonymous) @ index.js:115
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ index.js:1252
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:334
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:334
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:193
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:166
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:193
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:52
respond @ client.js:124
_RECEIVE_MESSAGE_TYPE.(anonymous function) @ types.js:111
receiveMessage @ index.js:112
messageListener @ index.js:138
postMessage (async)
(anonymous) @ strategies.js:56
SEND_MESSAGE_STRATEGIES.(anonymous function) @ strategies.js:55
(anonymous) @ index.js:72
ZalgoPromise.try @ promise.js:380
(anonymous) @ index.js:66
(anonymous) @ promise.js:357
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:357
ZalgoPromise.map @ promise.js:364
(anonymous) @ index.js:64
ZalgoPromise.try @ promise.js:380
sendMessage @ index.js:29
respond @ types.js:39
(anonymous) @ types.js:79
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:52
Promise rejected (async)
(anonymous) @ hacks.js:94
ZalgoPromise @ promise.js:42
(anonymous) @ hacks.js:57
ZalgoPromise.try @ promise.js:380
(anonymous) @ component.jsx:283
resultValue @ props.js:85
(anonymous) @ serialize.js:37
ZalgoPromise.try @ promise.js:380
(anonymous) @ serialize.js:36
(anonymous) @ types.js:65
ZalgoPromise.try @ promise.js:380
_RECEIVE_MESSAGE_TYPE.(anonymous function) @ types.js:53
receiveMessage @ index.js:112
messageListener @ index.js:138
postMessage (async)
(anonymous) @ strategies.js:56
SEND_MESSAGE_STRATEGIES.(anonymous function) @ strategies.js:55
(anonymous) @ index.js:72
ZalgoPromise.try @ promise.js:380
(anonymous) @ index.js:66
(anonymous) @ promise.js:357
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:357
ZalgoPromise.map @ promise.js:364
(anonymous) @ index.js:64
ZalgoPromise.try @ promise.js:380
sendMessage @ index.js:29
(anonymous) @ client.js:133
ZalgoPromise @ promise.js:42
(anonymous) @ client.js:111
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ client.js:109
ZalgoPromise.try @ promise.js:380
request @ client.js:34
_send @ client.js:211
wrapper @ serialize.js:155
(anonymous) @ promise.js:40
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:39
resultValue @ props.js:85
(anonymous) @ fn.js:69
queryParam @ component.js:200
(anonymous) @ props.js:181
ZalgoPromise.try @ promise.js:380
getQueryParam @ props.js:179
(anonymous) @ props.js:233
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ props.js:226
propsToQuery @ props.js:204
ParentComponent.buildUrl @ index.js:450
buildUrl:memoized @ decorators.js:13
(anonymous) @ index.js:190
ZalgoPromise.try @ promise.js:380
ParentComponent.tryInit @ index.js:1250
ParentComponent.render @ index.js:120
(anonymous) @ index.js:253
ZalgoPromise.try @ promise.js:380
ParentComponent.tryInit @ index.js:1250
ParentComponent.renderTo @ index.js:231
(anonymous) @ index.js:473
ZalgoPromise.try @ promise.js:380
Component.renderTo @ index.js:472
(anonymous) @ popupBridge.js:172
doRender @ popupBridge.js:162
Checkout.renderTo @ popupBridge.js:172
callOriginal @ util.js:279
(anonymous) @ hacks.js:131
obj.(anonymous function) @ util.js:275
renderCheckout @ checkout.js:80
clickButton @ button.js:48
(anonymous) @ button.js:63
index.js:574 ppxo_unhandled_error {stack: "Error: PayPal payment options are invalid.↵    at …://www.paypalobjects.com/api/checkout.js:2786:13)", errtype: "[object Error]", timestamp: 1508911256421, windowID: "75a7ba605c", pageID: "5d86480262", …}
window.console.window.console.(anonymous function) @ index.js:574
print @ logger.js:65
log @ logger.js:175
error @ logger.js:228
(anonymous) @ setup.js:33
dispatchPossiblyUnhandledError @ exceptions.js:18
(anonymous) @ promise.js:111
setTimeout (async)
ZalgoPromise.reject @ promise.js:109
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:334
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:166
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:193
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:52
respond @ client.js:124
_RECEIVE_MESSAGE_TYPE.(anonymous function) @ types.js:111
receiveMessage @ index.js:112
messageListener @ index.js:138
postMessage (async)
(anonymous) @ strategies.js:56
SEND_MESSAGE_STRATEGIES.(anonymous function) @ strategies.js:55
(anonymous) @ index.js:72
ZalgoPromise.try @ promise.js:380
(anonymous) @ index.js:66
(anonymous) @ promise.js:357
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:357
ZalgoPromise.map @ promise.js:364
(anonymous) @ index.js:64
ZalgoPromise.try @ promise.js:380
sendMessage @ index.js:29
respond @ types.js:39
(anonymous) @ types.js:79
(anonymous) @ promise.js:164
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:159
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.reject @ promise.js:116
(anonymous) @ promise.js:52
Promise rejected (async)
(anonymous) @ hacks.js:94
ZalgoPromise @ promise.js:42
(anonymous) @ hacks.js:57
ZalgoPromise.try @ promise.js:380
(anonymous) @ component.jsx:283
resultValue @ props.js:85
(anonymous) @ serialize.js:37
ZalgoPromise.try @ promise.js:380
(anonymous) @ serialize.js:36
(anonymous) @ types.js:65
ZalgoPromise.try @ promise.js:380
_RECEIVE_MESSAGE_TYPE.(anonymous function) @ types.js:53
receiveMessage @ index.js:112
messageListener @ index.js:138
postMessage (async)
(anonymous) @ strategies.js:56
SEND_MESSAGE_STRATEGIES.(anonymous function) @ strategies.js:55
(anonymous) @ index.js:72
ZalgoPromise.try @ promise.js:380
(anonymous) @ index.js:66
(anonymous) @ promise.js:357
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:357
ZalgoPromise.map @ promise.js:364
(anonymous) @ index.js:64
ZalgoPromise.try @ promise.js:380
sendMessage @ index.js:29
(anonymous) @ client.js:133
ZalgoPromise @ promise.js:42
(anonymous) @ client.js:111
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ client.js:109
ZalgoPromise.try @ promise.js:380
request @ client.js:34
_send @ client.js:211
wrapper @ serialize.js:155
(anonymous) @ promise.js:40
ZalgoPromise.try @ promise.js:380
(anonymous) @ promise.js:39
resultValue @ props.js:85
(anonymous) @ fn.js:69
queryParam @ component.js:200
(anonymous) @ props.js:181
ZalgoPromise.try @ promise.js:380
getQueryParam @ props.js:179
(anonymous) @ props.js:233
(anonymous) @ promise.js:150
ZalgoPromise.dispatch @ promise.js:198
ZalgoPromise.then @ promise.js:231
(anonymous) @ props.js:226
propsToQuery @ props.js:204
ParentComponent.buildUrl @ index.js:450
buildUrl:memoized @ decorators.js:13
(anonymous) @ index.js:190
ZalgoPromise.try @ promise.js:380
ParentComponent.tryInit @ index.js:1250
ParentComponent.render @ index.js:120
(anonymous) @ index.js:253
ZalgoPromise.try @ promise.js:380
ParentComponent.tryInit @ index.js:1250
ParentComponent.renderTo @ index.js:231
(anonymous) @ index.js:473
ZalgoPromise.try @ promise.js:380
Component.renderTo @ index.js:472
(anonymous) @ popupBridge.js:172
doRender @ popupBridge.js:162
Checkout.renderTo @ popupBridge.js:172
callOriginal @ util.js:279
(anonymous) @ hacks.js:131
obj.(anonymous function) @ util.js:275
renderCheckout @ checkout.js:80
clickButton @ button.js:48
(anonymous) @ button.js:63
types.js:111 Uncaught Error: PayPal payment options are invalid.
    at Object._RECEIVE_MESSAGE_TYPE.(anonymous function) [as postrobot_message_response] (https://www.paypalobjects.com/api/checkout.js:2863:118)
    at receiveMessage (https://www.paypalobjects.com/api/checkout.js:2765:77)
    at messageListener (https://www.paypalobjects.com/api/checkout.js:2786:13)`

So boo. 

My code is up at https://paypal.terrifickid.net also posted below.

Thanks!
TK

[code.txt](https://github.com/braintree/braintree-web/files/1413418/code.txt)

**Comment by crookedneighbor** [comment]
I was able to go through the complete login flow and get a nonce without issue on the demo site you provided. Are you still having issues?

**Comment by terrifickid** [comment]
Hello,

Yes I can confirm 422 response resolved. I contacted braintree support, they said:

"I took a look at your server logs and I can see that the error you're seeing is due to not having a PayPal billing agreement setup on your PayPal Sandbox account ___@___.com It is required to have a billing agreement setup when using the Braintree PayPal vault. 

I reached out to our PayPal representatives here to help enable that for you within your sandbox environment. This should allow you test successfully now. "

thanks for checking! great api!
